lib/iris/tests/unit/aux_factory/test_AuxCoordFactory.py

```diff
@@ -80,5 +80,82 @@ def test_lazy_complex(self):
         self.assertArrayEqual(result, expected)
 
 
+
+class Test__nd_bounds(tests.IrisTest):
+    def test_numpy_scalar_coord(self):
+        bounds = np.arange(2).reshape(1, 2)
+        coord = iris.coords.AuxCoord(np.arange(1), bounds=bounds)
+        result = AuxCoordFactory._nd_bounds(coord, (), 2)
+        expected = bounds[np.newaxis]
+        self.assertArrayEqual(result, expected)
+
+    def test_numpy_simple(self):
+        points = np.arange(12).reshape(4, 3)
+        bounds = np.arange(24).reshape(4, 3, 2)
+        coord = iris.coords.AuxCoord(points, bounds=bounds)
+        result = AuxCoordFactory._nd_bounds(coord, (0, 1), 2)
+        expected = bounds
+        self.assertArrayEqual(result, expected)
+
+    def test_numpy_complex(self):
+        points = np.arange(12).reshape(4, 3)
+        bounds = np.arange(24).reshape(4, 3, 2)
+        coord = iris.coords.AuxCoord(points, bounds=bounds)
+        result = AuxCoordFactory._nd_bounds(coord, (3, 2), 5)
+        expected = bounds.T[np.newaxis, np.newaxis, ..., np.newaxis]
+        self.assertArrayEqual(result, expected)
```

Comment:
@bjlittle This test fails due to a mismatch between the resultant and expected order of coordinates after transpose.